sapi/fpm/fpm/fpm_events.h

```diff
@@ -14,11 +14,15 @@ struct fpm_event_s {
 	int fd;                   /* not set with FPM_EV_TIMEOUT */
 	struct timeval timeout;   /* next time to trigger */
 	struct timeval frequency;
-	void (*callback)(struct fpm_event_s *, short, void *);
+	union {
+		void (*ptr_callback)(struct fpm_event_s *, short, void *);
+		void (*pid_callback)(struct fpm_event_s *, short, pid_t pid);
+	};
 	void *arg;
 	int flags;
 	int index;                /* index of the fd in the ufds array */
 	short which;              /* type of event */
+	pid_t pid;
```

Comment:
pid/arg could be union too ?

Reply:
How would you do the `if (ev->pid) { ... } else { ... }` part then?

Reply:
directly ? no. but there is ways to distinguish the two cases.

Reply:
> . I wanted to leave arg and pid separated for easier debugging but could be potentially changed to union if there are more types added.

That was my point indeed ok let's leave it as then.